### PR TITLE
Rename template parameters for clarity and consistency

### PR DESCRIPTION
**Pull request by witemple-msft**
This is in preparation for merging named template argument instantiation.

### CODE REVIEW

File: packages/http/lib/http.tsp [hunk 1]
```diff
@@ -21,16 +21,16 @@ model Response<Status> {
...
- * This can be useful in situations where you cannot use a bare T as the body
+ * This can be useful in situations where you cannot use a bare type as the body
  * and it is awkward to add a property.
  *
- * @template T The type of the model's `body` property.
+ * @template TBody The type of the model's `body` property.
  */
 @doc("")
-model Body<T> {
+model Body<TBody> {
```
Comment (timotheeguerin) [inline packages/http/lib/http.tsp:30]: Would `Type` would be better or something where we don't add the `T` prefix as this seems to be the pattern you have done

File: docs/libraries/http/reference/data-types.md [hunk 2]
```diff
@@ -97,18 +97,18 @@ model TypeSpec.Http.BearerAuth
...
 ```
 
 #### Template Parameters
 
-| Name | Description                              |
-| ---- | ---------------------------------------- |
-| T    | The type of the model's `body` property. |
+| Name  | Description                              |
+| ----- | ---------------------------------------- |
+| TBody | The type of the model's `body` property. |
```
Comment (tjprescott) [inline docs/libraries/http/reference/data-types.md:111]: It looks in other cases you dropped the T (TLocation -> Location) so why is this TBody?

File: docs/libraries/rest/reference/data-types.md [hunk 3]
```diff
@@ -21,102 +21,103 @@ scalar TypeSpec.Rest.ResourceLocation
...
-model TypeSpec.Rest.Resource.CollectionWithNextLink<T>
+model TypeSpec.Rest.Resource.CollectionWithNextLink<Item>
 ```
 
 #### Template Parameters
 
 | Name | Description                          |
 | ---- | ------------------------------------ |
-| T    | The resource type of the collection. |
+| Item | The resource type of the collection. |
```
Comment (tjprescott) [inline docs/libraries/rest/reference/data-types.md:36]: Why not just `Resource`?

File: docs/libraries/rest/reference/interfaces.md [hunk 4]
```diff
@@ -13,635 +13,635 @@ toc_max_heading_level: 3
...
+op TypeSpec.Rest.Resource.ExtensionResourceCollectionOperations.create(resource: TypeSpec.Rest.Resource.ResourceCreateModel<Resource>): Extension | TypeSpec.Rest.Resource.ResourceCreatedResponse<Resource> | Error
 ```
 
 #### `ExtensionResourceCollectionOperations.list` {#TypeSpec.Rest.Resource.ExtensionResourceCollectionOperations.list}
 
 Lists all instances of the extension resource.
 
 ```typespec
-op TypeSpec.Rest.Resource.ExtensionResourceCollectionOperations.list(): TypeSpec.Rest.Resource.CollectionWithNextLink<T> | TError
+op TypeSpec.Rest.Resource.ExtensionResourceCollectionOperations.list(): TypeSpec.Rest.Resource.CollectionWithNextLink<Item> | Error
```
Comment (tjprescott) [inline docs/libraries/rest/reference/interfaces.md:40]: Same here. I think the "Item" is a resource, so `Resource` makes sense here.

File: docs/libraries/rest/reference/interfaces.md [hunk 5]
```diff
@@ -13,635 +13,635 @@ toc_max_heading_level: 3
...
+op TypeSpec.Rest.Resource.ExtensionResourceOperations.create(resource: TypeSpec.Rest.Resource.ResourceCreateModel<Resource>): Extension | TypeSpec.Rest.Resource.ResourceCreatedResponse<Resource> | Error
 ```
 
 #### `ExtensionResourceOperations.list` {#TypeSpec.Rest.Resource.ExtensionResourceOperations.list}
 
 Lists all instances of the extension resource.
 
 ```typespec
-op TypeSpec.Rest.Resource.ExtensionResourceOperations.list(): TypeSpec.Rest.Resource.CollectionWithNextLink<T> | TError
+op TypeSpec.Rest.Resource.ExtensionResourceOperations.list(): TypeSpec.Rest.Resource.CollectionWithNextLink<Item> | Error
```
Comment (tjprescott) [inline docs/libraries/rest/reference/interfaces.md:232]: Rec: `Resource`

File: docs/libraries/rest/reference/interfaces.md [hunk 6]
```diff
@@ -13,635 +13,635 @@ toc_max_heading_level: 3
...
+op TypeSpec.Rest.Resource.ResourceCollectionOperations.create(resource: TypeSpec.Rest.Resource.ResourceCreateModel<Resource>): Resource | TypeSpec.Rest.Resource.ResourceCreatedResponse<Resource> | Error
 ```
 
 #### `ResourceCollectionOperations.list` {#TypeSpec.Rest.Resource.ResourceCollectionOperations.list}
 
 Lists all instances of the resource.
 
 ```typespec
-op TypeSpec.Rest.Resource.ResourceCollectionOperations.list(): TypeSpec.Rest.Resource.CollectionWithNextLink<T> | TError
+op TypeSpec.Rest.Resource.ResourceCollectionOperations.list(): TypeSpec.Rest.Resource.CollectionWithNextLink<Item> | Error
```
Comment (tjprescott) [inline docs/libraries/rest/reference/interfaces.md:311]: Rec: `Resource`

File: docs/libraries/rest/reference/interfaces.md [hunk 7]
```diff
@@ -13,635 +13,635 @@ toc_max_heading_level: 3
...
+| Resource | The resource model to list. |
+| Error    | The error response.         |
 
 #### `ResourceList.list` {#TypeSpec.Rest.Resource.ResourceList.list}
 
 Lists all instances of the resource.
 
 ```typespec
-op TypeSpec.Rest.Resource.ResourceList.list(): TypeSpec.Rest.Resource.CollectionWithNextLink<T> | TError
+op TypeSpec.Rest.Resource.ResourceList.list(): TypeSpec.Rest.Resource.CollectionWithNextLink<Item> | Error
```
Comment (tjprescott) [inline docs/libraries/rest/reference/interfaces.md:465]: Rec: `Resource`

File: docs/libraries/rest/reference/interfaces.md [hunk 8]
```diff
@@ -13,635 +13,635 @@ toc_max_heading_level: 3
...
+op TypeSpec.Rest.Resource.ResourceOperations.create(resource: TypeSpec.Rest.Resource.ResourceCreateModel<Resource>): Resource | TypeSpec.Rest.Resource.ResourceCreatedResponse<Resource> | Error
 ```
 
 #### `ResourceOperations.list` {#TypeSpec.Rest.Resource.ResourceOperations.list}
 
 Lists all instances of the resource.
 
 ```typespec
-op TypeSpec.Rest.Resource.ResourceOperations.list(): TypeSpec.Rest.Resource.CollectionWithNextLink<T> | TError
+op TypeSpec.Rest.Resource.ResourceOperations.list(): TypeSpec.Rest.Resource.CollectionWithNextLink<Item> | Error
```
Comment (tjprescott) [inline docs/libraries/rest/reference/interfaces.md:520]: Rec: `Resource`

File: docs/standard-library/built-in-data-types.md [hunk 9]
```diff
@@ -67,14 +67,14 @@ model OmitDefaults<T>
...
-model OmitProperties<T, TKeys>
+model OmitProperties<Source, Keys>
 ```
 
 #### Template Parameters
 | Name | Description |
 |------|-------------|
-| T | An object whose properties are spread. |
-| TKeys | The property keys to omit. |
+| Source | An object whose properties are spread. |
```
Comment (tjprescott) [inline docs/standard-library/built-in-data-types.md:76]: Perhaps in these scenarios `Target` would be more intuitve than `Source`?

File: docs/standard-library/built-in-data-types.md [hunk 10]
```diff
@@ -99,13 +99,13 @@ model OptionalProperties<T>
...
 ```typespec
-model Record<T>
+model Record<Property>
 ```
 
 #### Template Parameters
 | Name | Description |
 |------|-------------|
-| T | The type of the properties |
+| Property | The type of the properties |
```
Comment (tjprescott) [inline docs/standard-library/built-in-data-types.md:108]: Rec: `PropertyType`. I realize we don't use type anywhere else, but this make it seem like it's a property reference.

File: packages/compiler/lib/lib.tsp [hunk 11]
```diff
@@ -138,18 +138,18 @@ scalar unixTimestamp32 extends utcDateTime;
...
 /**
- * @dev Model with string properties where all the properties have type `T`
- * @template T The type of the properties
+ * @dev Model with string properties where all the properties have type `Property`
+ * @template Property The type of the properties
  */
-@indexer(string, T)
-model Record<T> {}
+@indexer(string, Property)
+model Record<Property> {}
```
Comment (tjprescott) [inline packages/compiler/lib/lib.tsp:152]: Consider `PropertyType`

Comment (timotheeguerin) [reply packages/compiler/lib/lib.tsp:152]: or `Value` as its a `Key, Value` thing

Comment (witemple-msft) [reply packages/compiler/lib/lib.tsp:152]: Will change to Element.

**Review by timotheeguerin — APPROVED** [review]

**Pull request merged**